scanner/src/funTest/kotlin/scanners/MultipleScannersTest.kt

```diff
@@ -0,0 +1,114 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner.scanners
+
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.matchers.shouldBe
+
+import org.ossreviewtoolkit.model.AnalyzerResult
+import org.ossreviewtoolkit.model.AnalyzerRun
+import org.ossreviewtoolkit.model.DependencyGraph
+import org.ossreviewtoolkit.model.DependencyGraphNode
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageLinkage
+import org.ossreviewtoolkit.model.PackageType
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.RootDependencyIndex
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.utils.test.shouldNotBeNull
+
+private const val DEFAULT_NS = "SpdxDocumentFile"
+private const val DEFAULT_VERSION = ""
+
+class MultipleScannersTest : WordSpec({
+    "Scanning a project and a package with overlapping provenance and non overlapping scanners" should {
+        val analyzerResult = createAnalyzerResult()
+
+        val dummyScanners = DummyScanner()
+        val ortResult = createScanner().copy(
+            scannerWrappers = mapOf(
+                PackageType.PROJECT to listOf(dummyScanners.copy(name = "Dummy2")),
+                PackageType.PACKAGE to listOf(dummyScanners)
+            )
+        ).scan(analyzerResult, skipExcluded = false, emptyMap())
+
+        "return scan results with non-overlapping scanners" {
+            ortResult.scanner.shouldNotBeNull {
+                val results = getScanResults(Identifier(DEFAULT_NS, "", "spdx-project", DEFAULT_VERSION))
+                results.map { it.scanner.name }.toSet() shouldBe setOf("Dummy2")
+                val results2 = getScanResults(Identifier(DEFAULT_NS, "", "spdx-dependency", DEFAULT_VERSION))
+                results2.map { it.scanner.name }.toSet() shouldBe setOf("Dummy")
+            }
+        }
+    }
+})
+
+private fun createAnalyzerResult(): OrtResult {
+    val vcsPackage = VcsInfo(
+        type = VcsType.GIT,
+        url = "https://github.com/nnobelis/spdx-dependency.git",
+        revision = "78c8a5b12ae2d0f613fff25f2d1eb746a768ece7",
+        path = ""
+    )
+
+    val pkg = Package.EMPTY.copy(
+        id = createId("spdx-dependency"),
+        vcs = vcsPackage,
+        vcsProcessed = vcsPackage.normalize()
+    )
+
+    val vcsProject = VcsInfo(
+        type = VcsType.GIT,
+        url = "https://github.com/nnobelis/spdx-project.git",
```

Comment:
This is using a repo under you namespace. I believe that you could just use any `test-data` repo under `oss-review-toolkit`. IIUC you are after overlapping provenances, not necessarily covering submodules. So, maybe https://github.com/oss-review-toolkit/ort-test-data-scanner can be used?

Reply:
I am really confused by this repository as there are no configuration file for any package managers ?!
How the project/package identifiers are supposed to look like ?

If I try to create a `Dummy::project:1.0.0` Project and  `Dummy::package:1.0.0` Package like you did in your `ScannerIntegrationFunTest`, the scanners are not workin.

I get a `ScannerDetails` with name `ProvenanceResolver` that contains a single issue:
`IOException: Could not resolve provenance for package 'Dummy::project:1.0.0' for source code origins [VCS, ARTIFACT].`.

Reply:
Yes I had a VCS info but in the end, the bug was caused by the dependency graph. When I removed it, the `ProvenanceResolver` error is gone.